citations-impl/impl/src/java/org/sakaiproject/citation/impl/BaseCitationService.java

```diff
@@ -2205,6 +2200,24 @@ else if(this.m_urls.containsKey(urlid))
 			}
 		}
 
+		@Override
+		public String getThumbnailURL() {
+			String imgURL = null;
+			try {
+				URL url = new URL(GOOGLE_BOOKS_SEARCH_API + "q=isbn:" + getCitationProperty("isnIdentifier"));
+				BufferedReader input = new BufferedReader(new InputStreamReader(url.openStream(), "UTF-8"));
```

Comment:
What's the timeout on the request? Can problems accessing the google API mean that the reading list fails to render?